### PR TITLE
campaigns: Add tests for executing ChangesetJobs

### PR DESCRIPTION
**Pull request by mrnugget**
This fixes https://github.com/sourcegraph/sourcegraph/issues/9103. It doesn't test every branch, but it is enough for now to give us more confidence in that function and to add more tests later.

### CODE REVIEW

File: enterprise/internal/campaigns/workers_test.go [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+					CommittedDate: now,
+				},
+			}},
+		},
+		CreatedAt: now,
+		UpdatedAt: now,
+	}
+}
+
+func buildBBSPR(now time.Time, c *cmpgn.Campaign, headRef string) interface{} {
```
Comment (unknwon) [inline enterprise/internal/campaigns/workers_test.go]: This name strikes me quite hard :)

Comment (mrnugget) [reply enterprise/internal/campaigns/workers_test.go]: Haha, I didn't even notice. Will fix.

**Comment by mrnugget** [comment]
Boom! Even found a bug: when the Changeset already exists in the database when we try to create it we didn't update the `CampaignIDs`.

**Comment by unknwon** [comment]
> Boom! Even found a bug: when the Changeset already exists in the database when we try to create it we didn't update the `CampaignIDs`.

image

Is this the one you just found? I'm experiencing this running Docker insider.

**Comment by eseliger** [comment]
Whoops that might have been me... which page are you on?

**Comment by mrnugget** [comment]
> Is this the one you just found? I'm experiencing this running Docker insider (after hit a retry when publish campaign failed).

No, your error is frontend-only. This here is "deep in the backend" and I can't easily think of a way to manually reproduce it without running into multiple failures :)

**Pull request merged**